### PR TITLE
Implement centered version of RMSProp optimizer

### PR DESCRIPTION
**Pull request by Lewuathe**
# Purpose

Centered version of RMSProp normalizes the update by the moving average of the gradient. RMSPropOptimizer can receive `centered` parameter like [rmsprop in tensorflow](https://github.com/tensorflow/tensorflow/blob/r1.5/tensorflow/python/training/rmsprop.py#L20).
This is a followup of https://github.com/PAIR-code/deeplearnjs/pull/687#issuecomment-364771968

see: https://github.com/PAIR-code/deeplearnjs/issues/729

---
This change is [Reviewable](https://reviewable.io/reviews/pair-code/deeplearnjs/880)

### CODE REVIEW

**Comment by Lewuathe** [comment]
@dsmilkov @nsthorat Could you take a look when you get a chance?

**Comment by nsthorat** [comment]
:lgtm_strong:

Nice work, implementation looks correct. One comment about code structure. Let me know what you think.

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[src/optimizers/rmsprop_optimizer.ts, line 110 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/880#-L9VihtO-Cd-Aqel517Z:-L9VihtO-Cd-Aqel517_:bxjx5zk) ([raw file](https://github.com/tensorflow/tfjs-core/blob/dade72563005df9b683c29768d01bc4a9a70104a/src/optimizers/rmsprop_optimizer.ts#L110)):*
> ```XML
>   }
> 
>   private applyGradientsWithPlain(variableGradients: NamedVariableMap) {
> ```

Nice work! This implementation looks correct (the TF comment [here](https://github.com/tensorflow/tensorflow/blob/r1.7/tensorflow/python/training/rmsprop.py#L33) is actually wrong).

I'm wondering, can we merge these methods? There is quite a bit in common, I think we can reduce the amount of code by merging with the bit flip inline.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/880#-:-L9VnVHUDnlWWmTP0cPP:bmirsfe)*

**Comment by nsthorat** [comment]
Nice work!
